src/api/apiErrors.js

```diff
@@ -11,7 +21,7 @@ import * as logging from '../utils/logging';
  * See docs: https://zulip.com/api/rest-error-handling
  */
 // TODO we currently raise these in more situations; fix that.
-export class ApiError extends Error {
+export class ApiError extends RequestError {
   code: ApiErrorCode;
```

Comment:
Could also have a commit that makes `ApiError` and `ServerError` read-only, right?

Reply:
Sure, yeah.

I guess when I did that on the base class:
```js
export class RequestError extends Error {
  +httpStatus: number | void;
  +data: mixed;
```
it was just because that's necessary in order for the subclasses to give more-specific types for those properties. But the same thing would be appropriate on the subclasses' properties too.